ambry-api/src/test/java/com.github.ambry/router/ByteBufferRSC.java

```diff
@@ -93,6 +105,34 @@ public long getSize() {
   }
 
   @Override
+  public void setDigestAlgorithm(String digestAlgorithm)
+      throws NoSuchAlgorithmException {
+    if (digest != null && digest.getAlgorithm().equals(digestAlgorithm)) {
```

Comment:
Don't want to recalculate if is the same `digestAlgorithm` as last time. Notice that `digestBytes` gets reset at the end of this function which triggers a recalculation on a call to `getDigest()`.